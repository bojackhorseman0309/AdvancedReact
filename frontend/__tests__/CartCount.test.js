import { render, screen } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });

  it('matches snapshot', () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });

  it('updates via props', async () => {
    const { container, rerender } = render(<CartCount count={11} />);
    expect(container.textContent).toBe('11');

    rerender(<CartCount count={12} />);

    expect(container.textContent).toBe('1211');
    await wait(400);
    // await screen.findByText('12');
    expect(container.textContent).toBe('12');
    expect(container).toMatchSnapshot();
  });
});
