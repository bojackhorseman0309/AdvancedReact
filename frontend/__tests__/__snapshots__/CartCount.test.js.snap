// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-dlbh5s-1 gOTkYw"
  >
    <div>
      <div
        class="CartCount__Dot-dlbh5s-0 fKpvyb count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-dlbh5s-1 gOTkYw"
  >
    <div>
      <div
        class="CartCount__Dot-dlbh5s-0 fKpvyb count count-enter-done"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
